front/lib/actions/mcp_internal_actions/servers/slack/helpers.ts

```diff
@@ -448,7 +463,234 @@ export async function hasSlackScope(
+      );
+    }
+
+    return new Err(new MCPError(`Channel not found: ${query}`));
+  }
+
+  // Search lookup.
+  // Scope="auto": search in joined channles; if no match fallback in all public channels.
+  if (scope === "auto") {
+    try {
+      const joinedChannels = await searchAndProcessChannels(
+        slackClient,
+        "joined",
+        query
+      );
+      if (joinedChannels.length > 0) {
+        return createChannelsResponse(
+          joinedChannels,
+          "in your joined channels"
+        );
+      }
+
+      // Fallback to all public channels.
+      const publicChannels = await searchAndProcessChannels(
+        slackClient,
+        "public",
+        query
+      );
+      return createChannelsResponse(
+        publicChannels,
+        "in public workspace channels"
+      );
+    } catch (error) {
+      return new Err(new MCPError(`Error searching channels: ${error}`));
+    }
+  }
+
+  // Scope="joined": search all public, private, im and mpim joined channels.
+  if (scope === "joined") {
+    try {
+      const joinedChannels = await searchAndProcessChannels(
+        slackClient,
+        "joined",
+        query
+      );
+      return createChannelsResponse(joinedChannels, "in your joined channels");
+    } catch (error) {
+      return new Err(new MCPError(`Error searching channels: ${error}`));
+    }
+  }
+
+  // Scope="all": search all public channels.
+  try {
+    const publicChannels = await searchAndProcessChannels(
+      slackClient,
+      "public",
+      query
+    );
+    return createChannelsResponse(
+      publicChannels,
+      "in public workspace channels"
+    );
+  } catch (error) {
+    return new Err(new MCPError(`Error searching channels: ${error}`));
+  }
+}
+
+// Used by slack_bot
+export async function executeListPublicChannels(
+  nameFilter: string | undefined,
+  accessToken: string,
+  mcpServerId: string
+) {
+  const slackClient = await getSlackClient(accessToken);
+  const channels = await getCachedPublicChannels({
+    mcpServerId,
+    slackClient,
+  });
+
+  if (nameFilter) {
+    const normalizedNameFilter = removeDiacritics(nameFilter.toLowerCase());
+    const filteredChannels = channels.filter(
+      (channel) =>
+        removeDiacritics(channel.name?.toLowerCase() ?? "").includes(
+          normalizedNameFilter
+        ) ||
+        removeDiacritics(channel.topic?.value?.toLowerCase() ?? "").includes(
+          normalizedNameFilter
+        )
+    );
+
+    // Early return if we found a channel.
+    if (filteredChannels.length > 0) {
+      return new Ok([
+        {
+          type: "text" as const,
+          text: `The workspace has ${filteredChannels.length} channels containing "${nameFilter}"`,
+        },
+        {
+          type: "text" as const,
+          text: JSON.stringify(
+            filteredChannels.map(cleanChannelPayload),
+            null,
+            2
+          ),
+        },
+      ]);
+    }
+  }
+  if (nameFilter) {
```

Comment:
This if is weirdly structured we have 

If(x) {}
if(x) {}

I undestand it might be correct but that's not well structured and hard to understand